src/nvim/cursor.c

```diff
@@ -182,6 +183,8 @@ static int coladvance2(
 
         ml_replace(pos->lnum, newline, false);
         changed_bytes(pos->lnum, (colnr_T)idx);
```

Comment:
use `inserted_bytes()` . also I think old size is zero and not idx?

Reply:
Should be addressed now, I also preventively fixed an issue with virtualedit in a TAB.